Analyticsv3Sample/AnalyticsSampleProject/DataGaSample.cs

```diff
@@ -0,0 +1,91 @@
+
+
+
+// Copyright 2015 DAIMTO :  www.daimto.com
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+// an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+//------------------------------------------------------------------------------
+// <auto-generated>
+//     This code was generated by DAIMTO-Google-apis-Sample-generator 1.0.0
+//     Build date: 11/17/2015 13:52:57
+//     C# generater version: 1.0.0
+//
+//     Changes to this file may cause incorrect behavior and will be lost if
+//     the code is regenerated.
+// </auto-generated>
+//------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Google.Apis.Analytics.v3;
+using Google.Apis.Analytics.v3.Data;
+
+namespace DAIMTO.Analytics.v3.Sample
+{
+    class DataGaSample
+    {
+        /// <summary>
+        /// Returns Analytics data for a view (profile).
+        /// Documentation: https://developers.google.com/analytics//v3/ga/get
+        /// </summary>
+        /// <param name="service">Valid authentcated AnalyticsService</param>
+        /// <param name="ids">Unique table ID for retrieving Analytics data. Table ID is of the form ga:XXXX, where XXXX is the Analytics view (profile) ID.</param>
+        /// <param name="startdate">Start date for fetching Analytics data. Requests can specify a start date formatted as YYYY-MM-DD, or as a relative date (e.g., today, yesterday, or 7daysAgo). The default value is 7daysAgo.</param>
+        /// <param name="enddate">End date for fetching Analytics data. Request can should specify an end date formatted as YYYY-MM-DD, or as a relative date (e.g., today, yesterday, or 7daysAgo). The default value is yesterday.</param>
+        /// <param name="metrics">A comma-separated list of Analytics metrics. E.g., 'ga:sessions,ga:pageviews'. At least one metric must be specified.</param>
+        /// <returns></returns>
+        public static GaData get(AnalyticsService service, string ids, string startdate, string enddate, string metrics)
+        {
+            try
+            {
+                var request = service.Data.Ga.Get(ids, startdate, enddate, metrics);
+                request.Dimensions = "ga:hour";
```

Comment:
You can provide another method that returns the request itself, then the user can add another optional paramters that he likes to add.
Since after that he expected to just call Execute or ExecuteAsync you don't need to provide another method that just calls it.